src/handlers/push_message.rs

```diff
@@ -54,13 +55,92 @@ pub async fn handler(
     headers: HeaderMap,
     RequireValidSignature(Json(body)): RequireValidSignature<Json<PushMessageBody>>,
 ) -> Result<Response> {
-    let request_id = get_req_id(&headers);
+    let (res, analytics_option) = handler_internal(
+        Path((tenant_id, id)),
+        StateExtractor(state),
+        headers,
+        RequireValidSignature(Json(body)),
+    )
+    .await;
 
-    increment_counter!(state.metrics, received_notifications);
+    let (status, response) = match res {
+        Ok(res) => (res.status_code.clone().as_u16(), res),
+        Err(error) => {
+            let res = error.into_response();
+            let status_code = res.status().clone().as_u16();
+            (status_code, res)
+        }
+    };
+
+    #[cfg(feature = "analytics")]
+    if let Some(mut message_info) = analytics_option {
+        message_info.status = status;
+        message_info.success = status >= 200 && status < 300;
```

Comment:
Nice to have, but isn't it enough with just `status` in the table?

Reply:
my intention with the success field was to give echo server the responsibility to determine when a request is successful or not, given it has more context on what's being done. Given it's just

>        message_info.success = status >= 200 && status < 300;

I don't mind dropping the column. 

Consider if in  the future  things becomes more complicated like `if status == 2xx AND some other condition` then it would be a problem for data to determine, with this column you control whether the outcome was successful or not.

Reply:
I've removed it for now